include/ddc/kernels/splines/spline_builder_batched.hpp

```diff
@@ -124,16 +128,58 @@ class SplineBuilderBatched
     template <class Layout>
     void operator()(
             ddc::ChunkSpan<double, spline_domain_type, Layout, memory_space> spline,
-            ddc::ChunkSpan<double, vals_domain_type, Layout, memory_space> vals) const;
+            ddc::ChunkSpan<double, vals_domain_type, Layout, memory_space> vals,
+            std::optional<ddc::ChunkSpan<double, derivs_domain_type, Layout, memory_space>> const
+                    derivs_xmin
+            = std::nullopt,
+            std::optional<ddc::ChunkSpan<double, derivs_domain_type, Layout, memory_space>> const
+                    derivs_xmax
+            = std::nullopt) const;
 };
 
 template <class SplineBuilder, class... IDimX>
 template <class Layout>
 void SplineBuilderBatched<SplineBuilder, IDimX...>::operator()(
         ddc::ChunkSpan<double, spline_domain_type, Layout, memory_space> spline,
-        ddc::ChunkSpan<double, vals_domain_type, Layout, memory_space> vals) const
+        ddc::ChunkSpan<double, vals_domain_type, Layout, memory_space> vals,
+        std::optional<ddc::ChunkSpan<double, derivs_domain_type, Layout, memory_space>> const
+                derivs_xmin,
+        std::optional<ddc::ChunkSpan<double, derivs_domain_type, Layout, memory_space>> const
+                derivs_xmax) const
 {
+    assert(vals.template extent<interpolation_mesh_type>()
+           == ddc::discrete_space<bsplines_type>().nbasis() - spline_builder.s_nbe_xmin
+                      - spline_builder.s_nbe_xmax);
+
+    const bool odd = spline_builder.s_odd;
+
     const std::size_t nbc_xmin = spline_builder.s_nbc_xmin;
+    const std::size_t nbc_xmax = spline_builder.s_nbc_xmax;
+
+    assert((BcXmin == ddc::BoundCond::HERMITE)
+           != (!derivs_xmin.has_value() || derivs_xmin->template extent<deriv_type>() == 0));
+    assert((BcXmax == ddc::BoundCond::HERMITE)
+           != (!derivs_xmax.has_value() || derivs_xmax->template extent<deriv_type>() == 0));
+
+    // Hermite boundary conditions at xmin, if any
+    // NOTE: For consistency with the linear system, the i-th derivative
+    //       provided by the user must be multiplied by dx^i
+    if constexpr (BcXmin == BoundCond::HERMITE) {
+        std::cout << derivs_xmin->template extent<deriv_type>() << "\n";
+        std::cout << nbc_xmin << "\n";
+        assert(derivs_xmin->template extent<deriv_type>() == nbc_xmin);
+        auto const dx_proxy = spline_builder.m_dx;
+        ddc::for_each(
+                ddc::policies::policy(exec_space()),
+                batch_domain(),
+                DDC_LAMBDA(typename batch_domain_type::discrete_element_type j) {
+                    for (int i = nbc_xmin; i > 0; --i) {
+                        spline(ddc::DiscreteElement<bsplines_type>(nbc_xmin - i), j)
+                                = (*derivs_xmin)(ddc::DiscreteElement<deriv_type>(i - 1), j)
```

Comment:
Should the `derivs_xmin` be saved to a local variable before the for_each to avoid incurring the cost of de-referencing the optional object inside the loop? Or do you expect the compiler to be smart enough to manage this itself?

Reply:
mmm that I do not know

Reply:
Yes maybe avoid optional in a kernel. The kernel is small, i do not expect it will impact much the performance.